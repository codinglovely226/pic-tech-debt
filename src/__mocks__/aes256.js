export default {
  encrypt: jest.fn(() => 'encryptedString'),
  decrypt: jest.fn(() => 'decryptedString'),
};
