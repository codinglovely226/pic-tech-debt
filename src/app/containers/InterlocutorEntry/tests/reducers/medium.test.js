/* eslint-disable */

import mediumReducer from '../../reducers/medium';
import aes256 from 'aes256';

import {
  updateInterlocutorMedium,
  updateInterlocutorMediumDetail,
} from '../../actions';

describe('mediumReducer', () => {
  const appKey = { id: 'appKeyId', key: 'appKey' };
  it('returns the initial state', () => {
    expect(mediumReducer(undefined, {})).toMatchSnapshot();
  });
  it('update caller medium', () => {
    expect(mediumReducer({}, updateInterlocutorMedium(1, 'Email'))).toMatchSnapshot();
  });
  it('update caller medium detail', () => {
    console.log(aes256.encrypt('key',' string'));
    expect(mediumReducer({}, updateInterlocutorMediumDetail(1, 'test@testing.test', appKey))).toMatchSnapshot();
  });
});
