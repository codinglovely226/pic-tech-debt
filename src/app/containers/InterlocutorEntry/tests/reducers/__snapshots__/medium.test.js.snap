exports[`mediumReducer returns the initial state 1`] = `
Object {
  "detail": "",
  "detailEnc": "",
  "type": "Phone",
}
`;

exports[`mediumReducer update caller medium 1`] = `
Object {
  "detail": "",
  "detailEnc": "",
  "type": "Email",
}
`;

exports[`mediumReducer update caller medium detail 1`] = `
Object {
  "detail": "test@testing.test",
  "detailEnc": "encryptedString",
  "type": "Phone",
}
`;
